packages/mux-player/src/time-display.ts

```diff
@@ -0,0 +1,69 @@
+const styles = `
+  :host {
+    cursor: pointer;
+  }
+`;
+
+const template = document.createElement("template");
+template.innerHTML = `
+  <style>
+    ${styles}
+  </style>
+  <media-time-display show-duration></media-time-display>
+`;
+
+const ButtonPressedKeys = ["Enter", " "];
+
+class MediaTimeDisplayToggle extends HTMLElement {
+  static styles: string = styles;
+  static template: HTMLTemplateElement = template;
+  timeDisplayEl: HTMLElement | null | undefined;
+
+  constructor() {
+    super();
+
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot?.appendChild(
+      (this.constructor as any).template.content.cloneNode(true)
+    );
+    this.timeDisplayEl = this.shadowRoot?.querySelector("media-time-display");
+  }
+
+  toggleTimeDisplay() {
+    if (this.timeDisplayEl?.hasAttribute("remaining")) {
+      this.timeDisplayEl?.removeAttribute("remaining");
+    } else {
+      this.timeDisplayEl?.setAttribute("remaining", "");
+    }
+  }
+
+  connectedCallback() {
+    const keyUpHandler = (e: KeyboardEvent) => {
+      const { key } = e;
+      if (!ButtonPressedKeys.includes(key)) {
+        this.removeEventListener("keyup", keyUpHandler);
+        return;
+      }
+
+      this.toggleTimeDisplay();
+    };
+
+    this.addEventListener("keydown", (e) => {
+      const { metaKey, altKey, key } = e;
+      if (metaKey || altKey || !ButtonPressedKeys.includes(key)) {
+        this.removeEventListener("keyup", keyUpHandler);
+        return;
+      }
+      this.addEventListener("keyup", keyUpHandler);
+    });
```

Comment:
borrowed this keyboard handling code (a11y requirement) from media-chrome-button

https://github.com/muxinc/media-chrome/blob/dd27a2a6620c804f51a6d496156805043edb6a44/src/js/media-chrome-button.js